packages/block-library/src/post-author/edit.js

```diff
@@ -1,31 +1,256 @@
+							onChange={ ( size ) => {
+								props.setAttributes( {
+									avatarSize: Number( size ),
+								} );
+							} }
+						/>
+					) }
+					<ToggleControl
+						label={ __( 'Show bio' ) }
+						checked={ showBio }
+						onChange={ () =>
+							props.setAttributes( { showBio: ! showBio } )
+						}
+					/>
+				</PanelBody>
+				<PanelBody title={ __( 'Text settings' ) }>
+					<FontSizePicker
+						value={ fontSize.size }
+						onChange={ setFontSize }
+					/>
+				</PanelBody>
+			</InspectorControls>
+
+			{ InspectorControlsColorPanel }
+
+			<BlockControls>
+				<AlignmentToolbar
+					value={ align }
+					onChange={ ( nextAlign ) => {
+						props.setAttributes( { align: nextAlign } );
+					} }
+				/>
+				<BlockColorsStyleSelector
+					TextColor={ TextColor }
+					BackgroundColor={ BackgroundColor }
+				>
+					{ ColorPanel }
+				</BlockColorsStyleSelector>
+			</BlockControls>
+
+			<TextColor>
+				<BackgroundColor>
+					<div
+						ref={ ref }
+						className={ classnames( blockClassNames, {
+							[ `has-text-align-${ align }` ]: align,
+						} ) }
+						style={ blockInlineStyles }
+					>
+						{ showAvatar && authorDetails && (
+							<div className="wp-block-post-author__avatar">
+								<img
+									width={ avatarSize }
+									src={
+										authorDetails.avatar_urls[ avatarSize ]
+									}
+									alt={ authorDetails.name }
+								/>
+							</div>
+						) }
+						<div className="wp-block-post-author__content">
+							{ ( ! RichText.isEmpty( byline ) ||
+								isSelected ) && (
+								<RichText
+									className="wp-block-post-author__byline"
+									multiline={ false }
+									placeholder={ __( 'Write byline …' ) }
+									withoutInteractiveFormatting
+									allowedFormats={ [
+										'core/bold',
+										'core/italic',
+										'core/strikethrough',
+									] }
+									value={ byline }
+									onChange={ ( value ) =>
+										props.setAttributes( { byline: value } )
+									}
+								/>
+							) }
+							<p className="wp-block-post-author__name">
+								{ authorDetails.name }
+							</p>
+							{ showBio && (
+								<p className={ 'wp-block-post-author__bio' }>
+									{ authorDetails.description }
+								</p>
+							) }
+						</div>
+					</div>
+				</BackgroundColor>
+			</TextColor>
+		</>
 	);
-	return author ? (
-		<address>
-			{ sprintf(
-				/* translators: %s: author name. */
-				__( 'By %s' ),
-				author.name
-			) }
-		</address>
-	) : null;
 }
 
-export default function PostAuthorEdit() {
-	if ( ! useEntityId( 'postType', 'post' ) ) {
+function PostAuthorEdit( props ) {
+	const [ postAuthor, setPostAuthor ] = useEntityProp(
+		'postType',
+		'post',
+		'author'
+	);
+
+	const [ authorDetails, setAuthorDetails ] = useState( false );
+
+	if ( ! postAuthor ) {
 		return 'Post Author Placeholder';
 	}
-	return <PostAuthorDisplay />;
+
+	if ( ! authorDetails ) {
+		getAuthorDetails( postAuthor, setAuthorDetails );
+	}
```

Comment:
I'm still a bit tempted to move all this into `PostAuthorDisplay` (group all the `use...` hooks before the first conditional `return`), and `PostAuthorEdit`, and rename `PostAuthorDisplay` to `PostAuthorEdit`. But we can do that in a follow-up.

Reply:
Yeah, but so many `use...` for the placeholder situation ...

Reply:
Yeah, that'd be a tradeoff. We can later experiment with that change and see how ugly it gets.